### ISSUE TITLE
Attempt to free memory that is still in use by an ongoing MPI communication

### ISSUE THREAD

**Issue by artpol84** [open-mpi/ompi#3268]
We observed this error only once in our MTT. We are running v2.x with SLURM/pmix there. It is possible that it is somehow related to this configuration, though I doubt that.
Here is the error message:

```
export OMPI_MCA_btl_openib_if_include=mlx4_0:1
OMPI_MCA_btl_openib_if_include=mlx4_0:1
OMPI_MCA_mpi_add_procs_cutoff=0+ OMPI_MCA_pmix_base_async_modex=1
OMPI_MCA_pmix_base_collect_data=0
/tmp/mtt_116453_slurm/bin/srun -N 8 -n 64 --mpi=pmix_v1 -p pmellanox <mtt-base>/installs/T8JL/tests/mpich_tests/mpich-mellanox.git/test/mpi/coll/allgather2
[boo13:10605] Attempt to free memory that is still in use by an ongoing MPI communication (buffer 0xa89000, size 9302016).  MPI job will now abort.
srun: error: boo13: task 9: Exited with exit code 1
srun: Terminating job step 1592.0
```

**Comment by artpol84** [comment]
**Update:** this has triggered today again.
First time I noticed that was on Thursday, 03/30/2017.

**Comment by artpol84** [comment]
This always happens for allgather2 test, so in conjunction to the time it first appeared and the commit history, I can say that this may be related to https://github.com/open-mpi/ompi/pull/3159

**Comment by artpol84** [comment]
@vspetrov

**Comment by hppritcha** [comment]
These MTT runs were done with vanilla Open MPI, no UCX, MXM, etc.

**Comment by gpaulsen** [comment]
From web-ex 04/11 - Not 100% reproducible, so lets keep this issue open a bit longer.

**Comment by artpol84** [comment]
still reproducible:
https://mtt.open-mpi.org/index.php?limit=&wrap=&trial=&enable_drilldowns=&yaxis_scale=&xaxis_scale=&hide_subtitle=&split_graphs=&remote_go=&do_cookies=&phase=test_run&text_start_timestamp=2017-04-11+19%3A49%3A53+-+2017-04-12+19%3A49%3A53&text_platform_hardware=%5Ex86_64%24&show_platform_hardware=show&text_os_name=%5ELinux%24&show_os_name=show&text_mpi_name=%5Eompi_vendor%24&show_mpi_name=show&text_mpi_version=%5Ev2.x%24&show_mpi_version=show&text_suite_name=%5Empich_tests%24&show_suite_name=show&text_test_name=all&show_test_name=hide&text_np=%5E32%24&show_np=show&text_full_command=&show_full_command=show&text_http_username=%5Emellanox%24&show_http_username=show&text_local_username=all&show_local_username=hide&text_platform_name=%5Emellanox-pmix%24&show_platform_name=show&click=Detail&phase=test_run&test_result=_run_f&text_compute_cluster_id=&text_os_version=&show_os_version=&text_platform_type=&show_platform_type=&text_submit_id=&text_hostname=&show_hostname=&text_mpi_get_id=&text_mpi_install_compiler_id=&text_mpi_install_configure_id=&text_mpi_install_id=&text_compiler_name=&show_compiler_name=&text_compiler_version=&show_compiler_version=&text_vpath_mode=&show_vpath_mode=&text_endian=&show_endian=&text_bitness=&show_bitness=&text_configure_arguments=&text_exit_value=&show_exit_value=&text_exit_signal=&show_exit_signal=&text_duration=&show_duration=&text_client_serial=&show_client_serial=&text_result_message=&text_result_stdout=&text_result_stderr=&text_environment=&text_description=&text_test_build_id=&text_test_build_compiler_id=&text_test_run_id=&text_test_name_id=&text_test_run_command_id=&text_test_suite_id=&text_performance_id=&text_launcher=&show_launcher=&text_resource_mgr=&show_resource_mgr=&text_network=&show_network=&text_parameters=&show_parameters=&lastgo=summary

**Comment by jsquyres** [comment]
Artem -- don't forget that you can click on the "Absolute date range" link on the right in MTT and get a short link: https://mtt.open-mpi.org/index.php?do_redir=2412

**Comment by artpol84** [comment]
Thanks, I didn't know that: https://mtt.open-mpi.org/index.php?do_redir=2412

**Comment by jsquyres** [comment]
@artpol84 Where are we on this issue?

**Comment by artpol84** [comment]
@jsquyres I don't see this error anymore in our MTT.

**Comment by jsquyres** [comment]
Sweet!  I'll close.

**Comment by lcebaman** [comment]
I am getting exactly this error with OpenMPI 3.0.0 and using RMA (multithreaded) in my code. This is reproducible  and it does not happen with MPICH. Any ideas?

**Comment by jsquyres** [comment]
@icebaman Can you open a new issue with a small reproducer code?

**Comment by chchang6** [comment]
Hi,
   We have a user who is running into this at the moment on our system, with OpenMPI 1.10.7 built against GCC 7.3.0. (Version 1 is because we are running into startup/hanging issues with newer versions at larger scales). Was this thread ever continued?

**Comment by jsquyres** [comment]
I'm afraid not.  Have you tried Open MPI v4.x?

**Comment by chchang6** [comment]
Hi Jeff, no but I also got that advice on another ticket for a different issue, so will try that shortly. Thanks!

**Comment by jsquyres** [comment]
Yeah, sorry, 1.10.x is ancient and not really supported any more.

**Comment by chchang6** [comment]
No worries; the only reason folks are running 1.10.X is that 2 and 3 are having scalability problems in our environment. Suspect it has something to do with startup mechanism, Slurm config, and our network, but hope 4 works out of the box.
